src/Sylius/Component/Core/Inventory/Updater/OnHandQuantityUpdater.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Inventory\Updater;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+final class OnHandQuantityUpdater implements OnHandQuantityUpdaterInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function decrease(OrderInterface $order)
+    {
+        $orderItems = $order->getItems();
+        foreach ($orderItems as $orderItem) {
+            Assert::greaterThanEq($orderItem->getQuantity(), 0, 'Quantity of order items cannot be negative number.');
```

Comment:
I think this validation is not the responsibility of this service, we should only validate that we can decrease the `onHand` by a certain amount (the second assert).